Source/Parallelization/WarpXComm.cpp

```diff
@@ -459,19 +543,25 @@ WarpX::FillBoundaryF (IntVect ng)
 void
 WarpX::FillBoundaryB_avg (IntVect ng)
 {
-    for (int lev = 0; lev <= finest_level; ++lev)
-    {
-        FillBoundaryB_avg(lev, ng);
-    }
+    // Launch asynchronous MPI communications
+    WarpX::FillBoundary_nowait( Bfield_avg_fp, ng, PatchType::fine );
+    WarpX::FillBoundary_nowait( Bfield_avg_cp, ng, PatchType::coarse );
+
+    // Finalize MPI communications
+    WarpX::FillBoundary_finish( Bfield_avg_fp, PatchType::fine );
+    WarpX::FillBoundary_finish( Bfield_avg_cp, PatchType::coarse );
 }
 
 void
 WarpX::FillBoundaryE_avg (IntVect ng)
 {
-    for (int lev = 0; lev <= finest_level; ++lev)
-    {
-        FillBoundaryE_avg(lev, ng);
-    }
+    // Launch asynchronous MPI communications
+    WarpX::FillBoundary_nowait( Efield_avg_fp, ng, PatchType::fine );
+    WarpX::FillBoundary_nowait( Efield_avg_cp, ng, PatchType::coarse );
```

Comment:
I think the comm for this was in the original code limited for` lev>0` on the coarse part:
`if (lev > 0) FillBoundaryE_avg(lev, PatchType::coarse, ng);`

Reply:
Yes, agreed, this is taken into account inside the `WarpX::FillBoundary_nowait` function (see line 496)